### PR TITLE
Artifact upload handling changed: temoporary file not used anymore

### PR DESCRIPTION
**Pull request by kjaskiewiczz**
Artifact upload handling changed.

Old approach:
- whole artifact file is uploaded to backed and saved in temporary file;
- artifact from temp file is uploaded to the file server.

New approach:
- metadata is read into local buffer;
- local buffer with metadata and rest of the uploaded file are uploaded to file server.

@bboozzoo @maciejmrowiec @mchalski

### CODE REVIEW

File: resources/images/s3/filestorage.go [hunk 1]
```diff
@@ -15,8 +15,10 @@
 package s3
 
 import (
+	//"bufio"
```
Comment (bboozzoo) [inline resources/images/s3/filestorage.go]: remove

Comment (kjaskiewiczz) [inline resources/images/s3/filestorage.go]: done

File: resources/images/s3/filestorage.go [hunk 2]
```diff
@@ -15,8 +15,10 @@
...
 	"fmt"
-	"os"
+	"io"
+	//"os"
```
Comment (kjaskiewiczz) [inline resources/images/s3/filestorage.go]: done

File: resources/images/controller/images_external_test.go [hunk 3]
```diff
@@ -463,7 +463,7 @@ func TestSoftwareImagesControllerNewImage(t *testing.T) {
 
 		model.On(
 			"CreateImage",
-			mock.AnythingOfType("*os.File"),
+			mock.Anything,
```
Comment (bboozzoo) [inline resources/images/controller/images_external_test.go]: I think it should be possible to use `mock.MatchedBy()` here, like this:

```
model.On(
    "CreateImage",
    mock.MatchedBy(func(ir interface{}) bool {
        _, ok := ir.(io.Reader)
        return ok
     }),
    ...
    )
```

care to try?

Comment (kjaskiewiczz) [inline resources/images/controller/images_external_test.go]: looks like it's working
updated

**Comment by bboozzoo** [comment]
@mendersoftware/rndity @maciejmrowiec anyone else wants to take a look?

**Pull request merged**